src/vm/gcinfodecoder.cpp

```diff
@@ -1539,8 +1539,7 @@ void GcInfoDecoder::ReportRegisterToGC(  // AMD64
     LOG((LF_GCROOTS, LL_INFO1000, "Reporting " FMT_REG, regNum ));
 
     OBJECTREF* pObjRef = GetRegisterSlot( regNum, pRD );
-
-#ifdef FEATURE_PAL
+#if defined(FEATURE_PAL) && !defined(SOS_TARGET_AMD64) 
```

Comment:
DACCESS_COMPILE isn’t defined for SOS.  It is just defined for the DAC (mscordaccore) and any code/libraries that are linked as part of the DAC.

So I agree that SOS_TARGET_AMD64 isn’t the best, but it works because gcinfodecoder.cpp isn’t used on x86 (not included by src\gcdump\i386\gcdumpx86.cpp).